File: prg1.cpp
#include<iostream.h>
#include<fstream.h>
#include<conio.h>
#include<string.h>
#include<process.h>
void main()
{
 fstream fp1,fp2;
 char fin[10],fout[10];
 int ch,n,i;
 char str[10],name[10][10];
 clrscr();
 for(;;)
 {
 cout<<"1:Std Input to Std Output 2:File to Std Output 3:File to File\n";
 cout<<"Enter your Choice : ";
 cin>>ch;
 switch(ch)
 {
 case 1: cout<<"Enter Number of Records : ";
 cin>>n;
 cout<<"Enter "<<n<<" Names : ";
 for(i=0;i<n;i++)
 {
 cin>>name[i];
 }
 cout<<"Reversed names are ... "<<endl;
 for(i=0;i<n;i++)
 {
 strrev(name[i]);
 cout<<name[i];
 cout<<"\n";
 }
 break;
case 2: cout<<"Enter the Input Filename : ";
 cin>>fin;
 fp1.open(fin,ios::in);
 while(!fp1.fail())
 {
 fp1>>str;
 strrev(str);
 cout<<str;
 cout<<"\n";
 }
 fp1.close();
 break;
 case 3: cout<<"Enter the Input Filename : ";
 cin>>fin;
 cout<<"Enter the Output Filename : ";
 cin>>fout;
 fp1.open(fin,ios::in);
 fp2.open(fout,ios::out);
 while(!fp1.fail())
 {
 fp1>>str;
 strrev(str);
 fp2<<str;
 fp2<<"\n";
 }
 fp1.close();
 fp2.close();
 break;
 default: exit(0);
 }
 }
}

File: prg1/prg1.py
def stdToStd():
    n = int(input("enter the number of records: "))
    arr = []

    print("enter names: ")
    for i in range(n):
        name = input()
        arr.append(name.strip())

    print("reversed names: ")
    for name in arr:
        print(name[::-1], end="\n")


def fileToFile():
    iFname = input("Enter the input file name: ")
    ifile = open(iFname, "r")
    names = [
        name.strip() for name in ifile.readlines()
    ]  # remove newline from the end of the records

    with open("outputFile.txt", "w") as f:
        for name in names:
            f.write(f"{name[::-1]}\n")


if __name__ == "__main__":
    while True:
        print("\n1. std input to std output\n2. file to file\n")
        ch = int(input("Enter your choice: "))
        if ch == 1:
            stdToStd()
        elif ch == 2:
            fileToFile()
        else:
            break
